### ISSUE TITLE
RedundantWithClauseResult false positive

### ISSUE THREAD

**Issue by fuelen** [rrrene/credo#885]
### Environment

* Credo version (`mix credo -v`): 1.6.0-dev-ref.master.6d195932
* Erlang/Elixir version (`elixir -v`): 1.12.2 (OTP 24.0..5)

### What were you trying to do?
```elixir
defmodule Test do
  def test(conn) do
    with ["Basic " <> token] <- get_req_header(conn, "authorization"),
         {:ok, decoded_token} <- token |> String.trim() |> Base.decode64(),
         [client_id, client_secret] <- String.split(decoded_token, ":"),
         {:ok, client} <- authorize(id: client_id, secret: client_secret) do
      {:ok, client}
    else
      _ -> :error
    end
  end

  def authorize(_params) do
    {:error, "ouch"}
  end
end
```

### Expected outcome
Don't apply `RedundantWithClauseResult` if `else` block is present

### Actual outcome
```
┃ [F] ↗ the last clause in `with` is redundant
┃       lib/test.ex:3 #(Test.test)
```

**Comment by NickNeck** [comment]
Good point. I will fix that. Thanks for reporting.

**Comment by rrrene** [comment]
@fuelen Nice catch! Thx for reporting! :+1:
